### DISCUSSION TITLE
weird indentation.

### DISCUSSION THREAD

**Discussion by zztczcx** [LunarVim/LunarVim#3785] [Q&A] [answered]
the indentation for ruby is a bit weird. it also happened when lsp is not started.

for example.
I am in a method.
```
def a
   self.b
end
```

when I type `self.`
this line will jump to the front. and became

```
def a
self.
end
```

not sure which caused this issue. so not sure if it due to plugin or config.

I have tried the pure clean nvim. which doesn't have this problem.

could someone help me?

**Comment by zztczcx** [comment, marked as answer]
figured it out.
it is caused by TreeSitter. 
I just disabled indent for ruby . 
now the indentation works well.

**Reply by AngelGallegosE** [reply]
There is an issue opened:
https://github.com/tree-sitter/tree-sitter-ruby/issues/230
